File: javascript/secret-handshake/secret-handshake.js
export const commands = (input) => {
  let output = []

  if (input & 0b00001) {
    output.push("wink")
  }

  if (input & 0b00010) {
    output.push("double blink")
  }

  if (input & 0b00100) {
    output.push("close your eyes")
  }

  if (input & 0b01000) {
    output.push("jump")
  }

  if (input & 0b10000) {
    output = output.reverse()
  }

  return output
}

File: cpp/secret-handshake/secret_handshake.cpp
#include "secret_handshake.h"
#include <algorithm>

namespace secret_handshake {
    vector<string> commands(int code) {
        vector<string> res;

        if ((code & 0b00001) != 0) {
            res.push_back("wink");
        }

        if ((code & 0b00010) != 0) {
            res.push_back("double blink");
        }

        if ((code & 0b00100) != 0) {
            res.push_back("close your eyes");
        }

        if ((code & 0b01000) != 0) {
            res.push_back("jump");
        }
        
        if ((code & 0b10000) != 0) {
            std::reverse(res.begin(), res.end());
        }

        return res;
    }
}  // namespace secret_handshake
